web/ui/react-app/src/pages/graph/GraphHelpers.ts

```diff
@@ -161,32 +161,54 @@ export const getColors = (data: { resultType: string; result: Array<{ metric: Me
   });
 };
 
-export const normalizeData = ({ queryParams, data }: GraphProps): GraphSeries[] => {
+export const normalizeData = ({ queryParams, data, exemplars, stacked }: GraphProps): GraphData => {
   const colors = getColors(data);
   const { startTime, endTime, resolution } = queryParams!;
-  return data.result.map(({ values, metric }, index) => {
-    // Insert nulls for all missing steps.
-    const data = [];
-    let pos = 0;
-
-    for (let t = startTime; t <= endTime; t += resolution) {
-      // Allow for floating point inaccuracy.
-      const currentValue = values[pos];
-      if (values.length > pos && currentValue[0] < t + resolution / 100) {
-        data.push([currentValue[0] * 1000, parseValue(currentValue[1])]);
-        pos++;
-      } else {
-        data.push([t * 1000, null]);
+  return {
+    series: data.result.map(({ values, metric }, index) => {
+      // Insert nulls for all missing steps.
+      const data = [];
+      let pos = 0;
+
+      for (let t = startTime; t <= endTime; t += resolution) {
+        // Allow for floating point inaccuracy.
+        const currentValue = values[pos];
+        if (values.length > pos && currentValue[0] < t + resolution / 100) {
+          data.push([currentValue[0] * 1000, parseValue(currentValue[1])]);
+          pos++;
+        } else {
+          data.push([t * 1000, null]);
+        }
       }
-    }
 
-    return {
-      labels: metric !== null ? metric : {},
-      color: colors[index].toString(),
-      data,
-      index,
-    };
-  });
+      return {
+        labels: metric !== null ? metric : {},
+        color: colors[index].toString(),
+        stack: stacked,
+        data,
+        index,
+      };
+    }),
+    exemplars: exemplars
+      ? exemplars
+          .map(({ seriesLabels, exemplars }) => {
+            const newLabels: { [key: string]: string } = {};
+            for (const label in seriesLabels) {
+              newLabels[`Series: ${label}`] = seriesLabels[label];
```

Comment:
Do we still need this merging of exemplar labels with series labels here now? I think it'd be cool to not prefix each of them with `Series: ` but modify the tooltip code in `GraphHelpers.tsx` to have a separate labels section that is only titled "Series labels:" once, and then lists the series labels separately.